runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobInvoker.java

```diff
@@ -64,6 +65,8 @@ public JobInvocation invoke(
     LOG.info("Invoking job {}", invocationId);
 
     flinkOptions.setFlinkMaster(firstNonNull(flinkOptions.getFlinkMaster(), flinkMasterUrl));
+    flinkOptions
```

Comment:
This overwrites filesToStage specified in the user's pipeline options with the classpath of job server.

- Are the user's filesToStage supposed to already have been staged by now so it's irrelevant?
- Who stages these filesToStage? It's definitely not the artifact staging service anymore, right?

Reply:
filesToStage is an option in FlinkPipelineOptions. With portability framework, the only file Flink is aware of is job sever jar and overwriting it.
* User files are staged using the beam artifact staging which is different.
* fileToStage are stage by flink and is in class path of flink task. Its not the beam artifact staging anymore.

Reply:
OK seems fine, please add a comment about this though.

Reply:
It's true that at this point we only need the core job server jar. However, I don't think this is the right place/way to do that. Instead, the job server driver should take the staging jar as a command line option (or a set of jars should be constructed from the classpath). Then, we should manually construct our own Flink `ExecutionEnvironment` rather than reuse the same method used by the non-portable runner. Sharing that code was a mistake.

Reply:
Specifically, I think it is incorrect to create an `ExecutionEnvironment` from pipeline options here:https://github.com/apache/beam/blob/34b071c0b0f08fceb451ce9c4a4bcaab6160abcd/runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java#L40

Reply:
I agree, I can change the code so that all the options should be provided explicitly to the method and read them in FlinkJobServiceDriver.